tests/test_ssl_transport_close.py

```diff
@@ -0,0 +1,83 @@
+from aiohttp import web
+import aiohttp
+import asyncio
+import warnings
+
+
+async def handle_root(request):
+    """
+    respond after server was restarted, forcing connection_lost
+    """
+    cq = request.app['cq']
+    dq = request.app['cq']
+    i = await cq.get()
+    cq.task_done()
+    await dq.put(i)
+    return web.Response(text='')
+
+
+async def _prepare(aiohttp_server, ssl_ctx, client_ssl_ctx, cq, dq):
+    app = web.Application()
+    app['cq'] = cq
+    app['dq'] = dq
+    app.router.add_get('/', handle_root)
+    server = await aiohttp_server(app, port=12345, ssl=ssl_ctx)
```

Comment:
It's better to use an ephemeral port 0. You can then get a client to whatever the actual port ends up being via [`aiohttp_client` fixture](https://github.com/aio-libs/aiohttp/blob/f8df85d/aiohttp/pytest_plugin.py#L334) and you don't need to care about managing the client session manually.